src/radical/pilot/agent/lm/yarn.py

```diff
@@ -160,7 +215,12 @@ def config_yarn_site():
             # Solution to find Java's home folder:
             # http://stackoverflow.com/questions/1117398/java-home-directory
 
-            jpos = subprocess.check_output(['readlink','-f', '/usr/bin/java']).split('bin')
+            java = subprocess.check_output(['which','java'])
+            if java != '/usr/bin/java':
+                jpos=java.split('bin')
+            else:
+                jpos = subprocess.check_output(['readlink','-f', '/usr/bin/java']).split('bin')
```

Comment:
I have to think a bit how to incorporate flag -f of the readlink. It makes the command to follow the symbolic link until it get to the actual executable. For example, if you do:

  readlink /usr/bin/java

on Ubuntu, it will return 

  /etc/alternatives/java

which is also a symlink.

I could do a while loop, but I am not sure it is going to be very nice.